### PR TITLE
remove orphan txn list from txpool

### PR DESCRIPTION
**Pull request by NoelBright**
remove orphan txn list from txpool, the txn witch has continuous
nonce can be append to txn pool, otherwise reject it.

Signed-off-by: NoelBright <noel.n.bright@gmail.com>

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [ ] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: chain/pool/txpool.go [hunk 1]
```diff
@@ -138,50 +110,44 @@ func (tp *TxnPool) processTx(txn *transaction.Transaction) error {
...
-				//TODO need more fee
-				isOrphan = false
-				replace = true
-			} else if list.Full() {
-				return errors.New("txpool is full")
-			} else if preNonce, _ := list.GetLatestNonce(); txn.UnsignedTx.Nonce == preNonce+1 {
-				isOrphan = false
+		if _, err := list.Get(txn.UnsignedTx.Nonce); err == nil {
+			log.Warning("replace old tx")
+			if err := tp.blockValidationState.VerifyTransactionWithBlock(txn, nil); err != nil {
```
Comment (yilunzhang) [inline chain/pool/txpool.go]: Should we replace the old txn in blockValidationState instead of trying to add it to blockValidationState? also @trueinsider

Comment (trueinsider) [reply chain/pool/txpool.go]: yeah, we should clean old tx before adding new one

File: chain/pool/txpool.go [hunk 2]
```diff
@@ -195,8 +161,9 @@ func (tp *TxnPool) GetAddressList() map[common.Uint160]int {
 	tp.TxLists.Range(func(k, v interface{}) bool {
 		if programHash, ok := k.(common.Uint160); ok {
 			if list, ok := v.(*NonceSortedTxs); ok {
-				count := len(list.txs) + len(list.orphans)
-				programHashes[programHash] = count
+				if len := list.Len(); len != 0 {
```
Comment (yilunzhang) [inline chain/pool/txpool.go]: we should avoid using reserved word "len"

File: chain/pool/txpool.go (hunk 1 above)
Comment (yilunzhang) [reply chain/pool/txpool.go]: We should also make sure that blockValidationState is not modified if processTx returns error

Comment (trueinsider) [reply chain/pool/txpool.go]: we can divide verify bvs into verify and append

Comment (trueinsider) [reply chain/pool/txpool.go]: or simply do clean in case of error

Comment (yilunzhang) [reply chain/pool/txpool.go]: Let's do this in a separate PR

**Pull request merged**